src/arrayMethodReduce.js

```diff
@@ -5,7 +5,22 @@
  */
 function applyCustomReduce() {
   [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+    let result;
+    let start;
+
+    if (arguments.length === 2) {
+      result = initialValue;
+      start = 0;
+    } else {
+      result = this[0];
+      start = 1;
+    }
```

Comment:
Then less code than better )

```suggestion
    let result = this[0];
    let start = 1;

    if (arguments.length === 2) {
      result = initialValue;
      start = 0;
    }
```